internal/core/data/rest_value_descriptor.go

```diff
@@ -0,0 +1,439 @@
+				http.Error(w, err.Error(), http.StatusNotFound)
+				return
+			default:
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+		}
+
+		pkg.Encode(v, w, LoggingClient)
+	}
+}
+
+// Return the value descriptors that are associated with a device
+// The value descriptor is expected parameters on puts or expected values on get/put commands
+// api/v1/valuedescriptor/devicename/{device}
+func valueDescriptorByDeviceHandler(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+
+	vars := mux.Vars(r)
+
+	device, err := url.QueryUnescape(vars["device"])
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		LoggingClient.Error("Error unescaping the device: " + err.Error())
+		return
+	}
+
+	ctx := r.Context()
+	// Get the value descriptors
+	vdList, err := getValueDescriptorsByDeviceName(device, ctx)
+	if err != nil {
+		switch err := err.(type) {
+		case *errors.ErrDbNotFound:
+			http.Error(w, err.Error(), http.StatusNotFound)
+			return
+		case *types.ErrServiceClient:
+			http.Error(w, err.Error(), err.StatusCode)
+		default:
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+	}
+
+	pkg.Encode(vdList, w, LoggingClient)
+}
+
+// Return the value descriptors that are associated with the device specified by the device ID
+// Associated value descripts are expected parameters of PUT commands and expected results of PUT/GET commands
+// api/v1/valuedescriptor/deviceid/{id}
+func valueDescriptorByDeviceIdHandler(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+
+	vars := mux.Vars(r)
+
+	deviceId, err := url.QueryUnescape(vars["id"])
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		LoggingClient.Error("Error unescaping the device ID: " + err.Error())
+		return
+	}
+
+	ctx := r.Context()
+	// Get the value descriptors
+	vdList, err := getValueDescriptorsByDeviceId(deviceId, ctx)
+	if err != nil {
+		switch err := err.(type) {
+		case *types.ErrServiceClient:
+			http.Error(w, err.Error(), err.StatusCode)
+		case *errors.ErrDbNotFound:
+			http.Error(w, err.Error(), http.StatusNotFound)
+			return
+		default:
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+	}
+
+	pkg.Encode(vdList, w, LoggingClient)
+}
+
+// restValueDescriptorsUsageHandler checks if value descriptors are currently being used.
+// This functionality is useful for determining if a value descriptor can be updated, or deleted.
+// This functionality does not provide any guarantee that the value descriptor will not be in use in the near future.
+// Any functionality using the check to perform updates or deletes is responsible for handling any race conditions which
+// may occur.
+// Returns a map[string]bool where the key is the ValueDescriptor Name and the value is a bool stating if the
+// ValueDescriptor is currently in use.
+func restValueDescriptorsUsageHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
I took the liberty of breaking up the `router.go` file to follow the other services. This function is the only thing that is new, everything else has been moved over from `router.go`.

Reply:
Yeah, I'm not %100 onboard with this. With this PR we've now got an inconsistency in core-data B/C value descriptors have their own implementation style that differs from events and readings. I see that you're going for a style similar to core-metadata, but now it's inconsistent until we create an issue to go back in and make it consistent. I wouldn't include this change in the PR.